optimum/habana/transformers/models/mpt/modeling_mpt.py

```diff
@@ -288,8 +454,10 @@ def prepare_inputs_for_generation(
         The only differences are:
         - add new args token_idx
         - add token_idx into model_inputs
+        - add reuse_cache
         - from step2 when enable KV cache, slice next_input_ids from input_ids base on the token_idx
         """
+        reuse_cache = kwargs.get("reuse_cache")
```

Comment:
```suggestion
        reuse_cache = kwargs.get("reuse_cache")
        cache_idx = kwargs.get("cache_idx")
```

be consistent, either do both of them like this or remove above line and use kwargs.get below

the other way is to use both as below,
```py
                "reuse_cache": kwargs.get("reuse_cache"),
                "cache_idx": kwargs.get("cache_idx"),
```